heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -315,18 +318,31 @@ public static void main(String[] args) throws Exception {
                 topologyPackage, topologyBinaryFile, topologyDefnFile, topology))
             .build());
 
-    LOG.fine("Static config loaded successfully ");
+    LOG.fine("Static config loaded successfully");
     LOG.fine(config.toString());
 
     SubmitterMain submitterMain = new SubmitterMain(config, topology);
-    boolean isSuccessful = submitterMain.submitTopology();
-
-    // Log the result and exit
-    if (!isSuccessful) {
-      throw new RuntimeException(String.format("Failed to submit topology %s", topology.getName()));
-    } else {
-      LOG.log(Level.FINE, "Topology {0} submitted successfully", topology.getName());
+    try {
+      submitterMain.submitTopology();
+      // SUPPRESS CHECKSTYLE IllegalCatch
+    } catch (Exception e) {
+      /* Since only stderr is used (by logging), we use stdout here to
```

Comment:
As implemented the user will always see the stack trace. If the output messaging is sufficient we could only show the stack trace with --verbose (i.e. log.fine). Thoughts?